include/tdi/common/tdi_utils.hpp

```diff
@@ -251,6 +252,167 @@ class TdiEndiannessHandler {
   }
 };
 
+namespace utils {
+
+class TableFieldUtils {
+ public:
+  TableFieldUtils() = delete;
+
+static inline tdi_status_t keyFieldSafeGet(
+      const tdi_id_t &field_id,
+      const tdi::KeyFieldInfo **key_field,
+      const tdi::KeyFieldValue *field_value,
+      const tdi::Table *table) {
+    *key_field = table->tableInfoGet()->keyFieldGet(field_id);
+    if (!key_field) {
+      LOG_ERROR("%s:%d %s Unable to find key for key field_id %d, ",
+                __func__,
+                __LINE__,
+                table->tableInfoGet()->nameGet().c_str(),
+                field_id);
+      return TDI_OBJECT_NOT_FOUND;
+    }
+
+    // If the key_field type is not what is required here, then just return
+    // error
+    if ((*key_field)->matchTypeGet() != field_value->matchTypeGet()) {
+      LOG_ERROR("%s:%d %s Incorrect key type provided for key field_id %d, ",
+                //        "expected type: %s received type: %s",
+                __func__,
+                __LINE__,
+                table->tableInfoGet()->nameGet().c_str(),
+                field_id
+                // KeyFieldTypeStr.at((*key_field)->dataTypeGet()),
+                // KeyFieldTypeStr.at(key_field_type_req)
+      );
+      return TDI_INVALID_ARG;
+    }
+    if ((*key_field)->isPtrGet() && field_value->size_ < 64) {
+      LOG_ERROR(
+          "%s:%d Field size is greater than 64 bits. Please use byte arrays in "
+          "the KeyFieldValue for field %d",
+          __func__,
+          __LINE__,
+          field_id);
+      return TDI_NOT_SUPPORTED;
+    }
+    return TDI_SUCCESS;
+  }
+
+  // Template classes must be defined whole in headers
+  template <class T>
```

Comment:
What classes are allowed for the template isn't clear. Let's put them in comments. It would be even better if they can instantiated over here only if the classes that work are public classes under tdi

Reply:
Updated the comment.
Also did the template functions Instantiation.